services/web/server/src/simcore_service_webserver/products/_api.py

```diff
@@ -48,7 +49,10 @@ async def get_current_product_credit_price_info(
     """
     current_product_name = get_product_name(request)
     repo = ProductRepository.create_from_request(request)
-    return await repo.get_product_latest_price_info_or_none(current_product_name)
+    return cast(
+        ProductPriceInfo | None,
+        await repo.get_product_latest_price_info_or_none(current_product_name),
+    )
```

Comment:
why is this cast needed here?? this repo is in the code right? this should be defined as the return type is it not?